src/isl/isl_type_reference.rs

```diff
@@ -35,6 +35,31 @@ impl IslTypeRef {
         IslTypeRef::Anonymous(IslTypeImpl::new(None, constraints.into()))
     }
 
+    /// Provides a string representing a definition of a nullable built in type reference
+    // these are the built types that rae annotated with `nullable`
+    fn get_nullable_type_reference_definition(type_ref_name: &str) -> IonSchemaResult<&str> {
+        Ok(match type_ref_name {
```

Comment:
There's a few more "core" types.

This should also include `lob`, `number`, `text`, and eventually `document`.

It should _maybe_ include `nothing` and `any` for completeness, but I'm not sure about them. What do you think?

Reply:
Sounds good. Will add it for `lob`, `number`, `text` and  a TODO for `document` (document type doesn't exist yet).
I will also add implementation of `nothing` with this PR and add it to this list for `nullable` annotation type references.

Reply:
So, I don't think the spec is clear whether `nullable::nothing` _should_ be allowed. (That's why I said I'm not sure about it.) 

I don't think that `nullable::nothing` and `nullable::any` are actually useful. In theory, `nullable::nothing` is equivalent to `$null`, and `nullable::any` is equivalent to `$any`. However, if the spec requires it, then we should include it whether or not we think it's useful.

In practice, the Kotlin implementation breaks for both `nullable::nothing` and `nullable::any`, when it encounters any typed null.

Reply:
Got it! I think in that case as the spec is not clear on this we can probably just include those eventhough they just point to `$null` or `$any` just to support it for all built in/ derived types.

Reply:
Sure.